api/app.go

```diff
@@ -332,7 +375,97 @@ func (app *App) Start() error {
 				zap.Int("graceperiod", graceperiod))
 		})
 		time.Sleep(time.Duration(graceperiod) * time.Millisecond)
+	case err := <-errch:
+		return err
+	case <-stopped:
 	}
 	log.I(l, "app stopped")
 	return nil
 }
+
+func (app *App) startGRPCServer(lis net.Listener, errch chan<- error) {
+	var basicAuthInterceptor grpc.UnaryServerInterceptor
+
+	basicAuthUser := app.Config.GetString("basicauth.username")
+	if basicAuthUser == "" {
+		basicAuthInterceptor = grpc.UnaryServerInterceptor(app.noAuthMiddleware)
+	} else {
+		basicAuthInterceptor = grpc_auth.UnaryServerInterceptor(app.basicAuthMiddleware)
+	}
+
+	app.grpcServer = grpc.NewServer(grpc.UnaryInterceptor(
+		grpc_middleware.ChainUnaryServer(
+			basicAuthInterceptor,
+			grpc.UnaryServerInterceptor(app.loggerMiddleware),
+			grpc.UnaryServerInterceptor(app.recoveryMiddleware),
+			grpc.UnaryServerInterceptor(app.responseTimeMetricsMiddleware),
+			grpc.UnaryServerInterceptor(app.sentryMiddleware),
+			grpc.UnaryServerInterceptor(app.newRelicMiddleware),
+		),
+	))
+	api.RegisterPodiumServer(app.grpcServer, app)
```

Comment:
Suggestion: some of the configuration setup could live in NewApp / Configure, which may be for the best in case some of it errors (see comment below).

Reply:
Will evaluate comments below.